Add support for ignoring times (`12:34`) `ignoreDigits` is on

danielgolden opened the pull request:
<!--
  Please check the needed checkboxes ([ ] -> [x]). Leave the
  comments as they are, they won’t show on GitHub.
  We are excited about pull requests, but please try to limit the scope, provide
  a general description of the changes, and remember, it’s up to you to convince
  us to land it.
-->

### Initial checklist

*   [x] I read the support docs <!-- https://github.com/retextjs/.github/blob/main/support.md -->
*   [x] I read the contributing guide <!-- https://github.com/retextjs/.github/blob/main/contributing.md -->
*   [x] I agree to follow the code of conduct <!-- https://github.com/retextjs/.github/blob/main/code-of-conduct.md -->
*   [x] I searched issues and couldn’t find anything (or linked relevant results below) <!-- https://github.com/search?q=user%3Aretextjs&type=Issues -->
*   [x] If applicable, I’ve added docs and tests

### Description of changes
- Adds a check in the `ignoreDigits` option which will ignore any word that is a time (e.g. `2:41pm`)
- Adds a new test to ensure that times are ignored.
- Updates README to include the updated documentation for the `ignoreDigits` option.

---

Closes #24 and a step toward resolving newrelic/one-core-toolbox#38

<!--do not edit: pr-->

danielgolden committed:
chore(Tests): Add test for ignore words with digits option

danielgolden committed:
chore(Tests): Improve existing test and add another

danielgolden committed:
chore(Options): Add  option
- It's set to  by default
- The JSDoc documentation for it has also been added

danielgolden committed:
docs(README): Add documentation for ignoreAnyDigits option

danielgolden committed:
docs(README): Minor formatting adjustment

wooorm reviewed (commented):
Thanks for working on this! ✨

What you implemented is a bit different from what the original issue was (“times in common formats (e.g. 2:41PM)”).
There is a difference between:

- finite words, which also include numbers, and can accidentally include numbers (`w0rd`, because the `o` is next to `0`), which have a great solution already: the personal dictionary.
- numerical-like, such as times, which don’t have a good solution.

Are there other words you want to ignore? Did you consider the personal dictionary (or `ignore` if you have to)?

I was thinking more of adding `/^\d{1,2}:\d{2}(?:[ap]\.?m)?$/i`. Either under the existing `ignoreDigits` option. Or, if you like it, perhaps a new `ignoreTimes` option?

We could also add the resultions you show in a test case?

danielgolden commented:
That's true, and a good point. There aren't really other words I want to ignore (times are my priority), they just seemed to fit in the same category as I was thinking about it (but I supposed this is exactly how scope creep happens haha).

> I was thinking more of adding /^\d{1,2}:\d{2}(?:[ap]\.?m)?$/i. Either under the existing ignoreDigits option

I really like this solution. Keeps the API simple and it fit's well the specific problem of ignoring times. I'll update the PR to reflect these changes.

Thanks for the feedback! :)

danielgolden committed:
chore(ignoreDigits): Strip back scope of new functionality
- Removes the recently added prop  and stuffs
it's time ignoring functionality into the  prop.
- Updates tests to reflect the above changes
- Updates README to reflect the above changes

danielgolden commented:
I'm not confident about the documentation updates. Is this a satisfactory way to document the ignoring of times? If so, awesome. If not, would love your opinion.

https://github.com/retextjs/retext-spell/blob/d71c22bf2917b28ade1157eb2c530e9a35002614/index.js#L18-L19
https://github.com/retextjs/retext-spell/blob/d71c22bf2917b28ade1157eb2c530e9a35002614/readme.md?plain=1#L112-L113

wooorm commented on index.js:
```diff
@@ -286,6 +286,10 @@ function all(tree, file, config) {
+    return (
+      ignore.includes(word) ||
+      (ignoreDigits && /^\d+$/.test(word)) ||
+      (ignoreDigits && /^\d{1,2}:\d{2}(?:[ap]\.?m)?$/i.test(word))
```
```suggestion
      (ignoreDigits && /^\d{1,2}:\d{2}(?:[ap]\.?m\.?)?$/i.test(word))
```

I think I missed this dot earlier that might be there?

wooorm commented on test.js:
```diff
@@ -223,6 +223,17 @@ test('should ignore digits', (t) => {
+
+  retext()
+    .use(retextSpell, enGb)
+    .process('2:41pm')
```
```suggestion
    .process('Let’s meet at 2:41pm.')
```

A more real example.

wooorm commented on test.js:
```diff
@@ -223,6 +223,17 @@ test('should ignore digits', (t) => {
 })
 
+test('should ignore times', (t) => {
+  t.plan(1)
```
```suggestion
  t.plan(2)
```

Let’s add another test.

wooorm commented on test.js:
```diff
@@ -223,6 +223,17 @@ test('should ignore digits', (t) => {
+    .process('2:41pm')
+    .then((file) => {
+      check(t, file, [])
+    }, t.ifErr)
```
```suggestion
    }, t.ifErr)

  retext()
    .use(retextSpell, enGb)
    .process("On my way! ETA 11:50!")
    .then((file) => {
      check(t, file, []);
    }, t.ifErr);
```

Another case for just a time

wooorm reviewed (commented):
Docs look good!

wooorm committed:
Apply suggestions from code review

wooorm commented:
Thanks, released in [`5.3.0`](https://github.com/retextjs/retext-spell/releases/tag/5.3.0)!

danielgolden commented:
Awesome! Thanks so much :)
